File: java_solutions/1464.maximum-product-of-two-elements-in-an-array.java
/*
 * @lc app=leetcode id=1464 lang=java
 *
 * [1464] Maximum Product of Two Elements in an Array
 */

class Solution {
    public int maxProduct(int[] nums) {
        return Arrays.stream(nums)
            .boxed()
            .sorted(Comparator.reverseOrder())
            .limit(2)
            .reduce((a, b) -> (a - 1) * (b - 1))
            .get();
    }
}

File: kotlin_solutions_1/1464.maximum-product-of-two-elements-in-an-array.kt
/*
 * @lc app=leetcode id=1464 lang=kotlin
 *
 * [1464] Maximum Product of Two Elements in an Array
 */

class Solution {
    fun maxProduct(nums: IntArray): Int = nums.sortedDescending().let { (it[0] - 1) * (it[1] - 1) }
}